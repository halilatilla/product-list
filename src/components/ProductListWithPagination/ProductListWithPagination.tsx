import { FC, useState } from 'react'

import { useAppSelector } from '@src/store'
import { Pagination, ProductList } from '@src/components'
import { paginationOptions } from '@src/constants'
import IProductListWithPagination from './ProductListWithPagination.types'
import styles from './ProductListWithPagination.module.scss'

const ProductListWithPagination: FC<IProductListWithPagination> = () => {
  const { filteredProducts } = useAppSelector((state) => state.filter)
  const [page, setPage] = useState(paginationOptions.START_PAGE)

  const handlePagination = (e: number) => {
    setPage(e)
  }

  return (
    <div>
      <ProductList page={page} products={filteredProducts} />
      <div className={styles.pagination}>
        <Pagination
          current={page}
          total={filteredProducts.length}
          pageSize={paginationOptions.PAGE_SIZE}
          onChange={handlePagination}
          hideOnSinglePage={true}
        />
      </div>
    </div>
  )
}

export default ProductListWithPagination
