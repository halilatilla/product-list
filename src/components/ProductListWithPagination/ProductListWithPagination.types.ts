export default interface IProductListWithPagination {
  className?: string
}
