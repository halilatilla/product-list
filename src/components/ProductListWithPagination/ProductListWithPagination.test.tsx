import { render, screen, within } from '@src/testUtils'

import ProductListWithPagination from './ProductListWithPagination'

describe('ProductListWithPagination', () => {
  it('renders correctly', () => {
    const { container } = render(<ProductListWithPagination />)
    expect(container).toMatchSnapshot()
  })
})
