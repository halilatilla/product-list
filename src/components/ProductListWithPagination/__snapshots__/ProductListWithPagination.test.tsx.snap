// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListWithPagination renders correctly 1`] = `
<div>
  <div>
    <ul
      aria-label="product-list"
      class="productList"
    />
    <div
      class="pagination"
    />
  </div>
</div>
`;
